From 6a75185ad4f5300848ed28f3b5ec792dd628d21a Mon Sep 17 00:00:00 2001
From: rnsbox <rnsbox@localhost>
Date: Thu, 15 Oct 2026 04:08:50 +0300
Subject: [PATCH 1/2] ramdisk: cv180x ramboot/onekernel use the minimal musl
 cpio like sg200x

The cv180x ramboot_fixed_files.txt + onekernel_fixed_files.txt.sqsh were
the old full-recovery recipes (inittab, mdev.conf, autologin, rcS,
init.sh.sqsh, bin/busybox, usr/sbin/blkid...) - files the musl_riscv64
initramfs base and overlay do not provide, so build_ramboot/pack_boot
die in gen_init_cpio on a fresh tree (Sipeed only exercised cv180x with
the glibc-era flow; the sg200x configs were already minimal).

Replace both with the sg200x minimal set: init/busybox/e2fsck/parted
straight from the musl base - all present. The base's recovery init
mounts /dev/mmcblk0p1 (boot) and switch_roots into mmcblk0p2, exactly
the cv1800b SD layout; the sg200x-panel msc() bits fail harmlessly.
ramboot.itb is only the USB-recovery image; the normal SD boot path is
boot.itb (SKIP_RAMDISK), where the cpio is generated but not embedded.
---
 .../cv180x/onekernel_fixed_files.txt.sqsh     | 50 +++++++------
 .../configs/cv180x/ramboot_fixed_files.txt    | 72 +++++--------------
 2 files changed, 42 insertions(+), 80 deletions(-)

diff --git a/ramdisk/configs/cv180x/onekernel_fixed_files.txt.sqsh b/ramdisk/configs/cv180x/onekernel_fixed_files.txt.sqsh
index a7a18beab..0b9559238 100644
--- a/ramdisk/configs/cv180x/onekernel_fixed_files.txt.sqsh
+++ b/ramdisk/configs/cv180x/onekernel_fixed_files.txt.sqsh
@@ -1,33 +1,31 @@
 
 #####################
-# files to boot a real root system
-
-file /init ../target/init.sh.sqsh 775 0 0
-file /init_recovery ../target/init.sh.recoverysh 775 0 0
+# files to boot a minimal system
 
 dir /dev 755 0 0
-dir /etc 755 0 0
-dir /proc 755 0 0
-dir /sys 755 0 0
-dir /mnt 755 0 0
-
+dir /dev/shm 755 0 0
+dir /dev/pts 755 0 0
 nod /dev/console 644 0 0 c 5 1
 
-dir /root 755 0 0
-dir /home 755 0 0
-dir /home/cvitek 755 1000 1000
-file /root/cmdline.sh ../target/cmdline.sh 775 0 0
+file /init ../target/init 775 0 0
+file /busybox ../target/busybox 775 0 0
+file /e2fsck ../target/e2fsck 775 0 0
+file /resize2fs ../target/resize2fs 775 0 0
+file /parted ../target/parted 775 0 0
+file /libc.so ../target/libc.so 775 0 0
+
+dir /proc 755 0 0
+dir /sys 755 0 0
 
-dir /bin 755 1000 1000
-dir /usr 755 1000 1000
-dir /usr/sbin 755 1000 1000
-file /bin/busybox ../target/bin/busybox 755 0 0
-file /usr/sbin/blkid ../target/usr/sbin/blkid 755 0 0
-slink /bin/sh /bin/busybox 755 0 0
-slink /bin/su /bin/busybox 755 0 0
-slink /bin/mount /bin/busybox 755 0 0
-slink /bin/umount /bin/busybox 755 0 0
-slink /bin/findfs /bin/busybox 755 0 0
-slink /bin/chown /bin/busybox 755 0 0
-slink /bin/chmod /bin/busybox 755 0 0
-slink /bin/switch_root /bin/busybox 755 0 0
+slink /sh /busybox 755 0 0
+slink /mount /busybox 755 0 0
+slink /ls /busybox 755 0 0
+slink /mkdir /busybox 755 0 0
+slink /sync /busybox 755 0 0
+slink /ln /busybox 755 0 0
+slink /cat /busybox 755 0 0
+slink /switch_root /busybox 755 0 0
+slink /grep /busybox 755 0 0
+slink /wc /busybox 755 0 0
+slink /umount /busybox 755 0 0
+slink /sleep /busybox 755 0 0
diff --git a/ramdisk/configs/cv180x/ramboot_fixed_files.txt b/ramdisk/configs/cv180x/ramboot_fixed_files.txt
index c7536f2c6..0b9559238 100644
--- a/ramdisk/configs/cv180x/ramboot_fixed_files.txt
+++ b/ramdisk/configs/cv180x/ramboot_fixed_files.txt
@@ -2,66 +2,30 @@
 #####################
 # files to boot a minimal system
 
-file /init ../target/init_ramboot.sh 775 0 0
-
 dir /dev 755 0 0
 dir /dev/shm 755 0 0
 dir /dev/pts 755 0 0
 nod /dev/console 644 0 0 c 5 1
 
-dir /bin 755 1000 1000
-file /bin/busybox ../target/bin/busybox 755 0 0
-file /bin/su ../target/bin/busybox 4755 0 0
-slink /bin/sh /bin/busybox 755 0 0
-slink /linuxrc /bin/busybox 755 0 0
-
+file /init ../target/init 775 0 0
+file /busybox ../target/busybox 775 0 0
+file /e2fsck ../target/e2fsck 775 0 0
+file /resize2fs ../target/resize2fs 775 0 0
+file /parted ../target/parted 775 0 0
+file /libc.so ../target/libc.so 775 0 0
 
-
-dir /sbin 755 0 0
 dir /proc 755 0 0
 dir /sys 755 0 0
-dir /mnt 755 0 0
-dir /run 755 0 0
-dir /tmp 755 0 0
-
-dir /var 755 0 0
-dir /var/run 755 0 0
-dir /var/log 755 0 0
-dir /var/empty 755 0 0
-
-dir /usr 755 0 0
-dir /usr/lib 755 0 0
-dir /usr/bin 755 0 0
-dir /usr/sbin 755 0 0
-dir /usr/local 755 0 0
-dir /usr/local/bin 755 0 0
-file /usr/local/bin/autologin ../target/usr/local/bin/autologin 755 1000 1000
-
-dir /root 755 0 0
-dir /home 755 0 0
-dir /home/cvitek 755 1000 1000
-
-dir /etc 755 0 0
-dir /etc/init.d 755 0 0
-dir /etc/network 755 1000 1000
-dir /etc/network/if-pre-up.d 755 1000 1000
-dir /etc/network/if-up.d 755 1000 1000
-dir /etc/network/if-down.d 755 1000 1000
-dir /etc/network/if-post-down.d 755 1000 1000
-file /etc/inittab ../target/etc/inittab 644 0 0
-file /etc/fstab ../target/etc/fstab 644 0 0
-file /etc/group ../target/etc/group 644 0 0
-file /etc/passwd ../target/etc/passwd 644 0 0
-file /etc/shadow ../target/etc/shadow 644 0 0
-file /etc/hostname ../target/etc/hostname 644 0 0
-file /etc/issue ../target/etc/issue 644 0 0
-file /etc/os-release ../target/etc/os-release 644 0 0
-file /etc/run_usb.sh ../target/etc/run_usb.sh 775 1000 1000
-file /etc/uhubon.sh ../target/etc/uhubon.sh 775 1000 1000
-file /etc/init.d/rcK ../target/etc/init.d/rcK 775 1000 1000
-file /etc/init.d/rcS ../target/etc/init.d/rcS 755 0 0
-file /etc/init.d/S00board ../target/etc/init.d/S00board 755 0 0
-file /etc/mdev.conf ../target/etc/mdev.conf 644 0 0
-
-dir /lib 755 0 0
 
+slink /sh /busybox 755 0 0
+slink /mount /busybox 755 0 0
+slink /ls /busybox 755 0 0
+slink /mkdir /busybox 755 0 0
+slink /sync /busybox 755 0 0
+slink /ln /busybox 755 0 0
+slink /cat /busybox 755 0 0
+slink /switch_root /busybox 755 0 0
+slink /grep /busybox 755 0 0
+slink /wc /busybox 755 0 0
+slink /umount /busybox 755 0 0
+slink /sleep /busybox 755 0 0
-- 
2.43.0

